pkg-manager/plugin-commands-installation/src/link.ts

```diff
@@ -109,11 +112,24 @@ export async function handler (
   let workspacePackagesArr: Project[]
   let workspacePackages!: WorkspacePackages
   if (opts.workspaceDir) {
-    workspacePackagesArr = await findWorkspacePackages(opts.workspaceDir, {
-      ...opts,
-      patterns: opts.workspacePackagePatterns,
-    })
-    workspacePackages = arrayOfWorkspacePackagesToMap(workspacePackagesArr) as WorkspacePackages
+    // https://github.com/orgs/pnpm/discussions/9037
+    // The pnpm-workspace.yaml file may only contain other configurations without the packages field.
+    const workspaceManifest = await readWorkspaceManifest(opts.workspaceDir)
+    if (!workspaceManifest?.packages && (params == null || params.length === 0)) {
```

Comment:
it shouldn't matter if it is a multi-package workspace or single-package one. It should always work the same and link to global.